Bottom: 098c4c0905ca654f12792d50c7b4c1c3e163285e
Top:    b104109bec01c4c173f369f93dc54eea82db6eb5
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-17 16:47:42 -0500

Add vmdaedalus inventory

Run ansible provisioning through Docker

Add 'restore' tag to tasks. By tagging the tasks, restoration can be skipped using --skip-tags
option to ansible-playbook. This is useful for testing ansible plays
on systems without tapedrives or established backups.

Add Vagrant configuration

Set docker container restart_policy to no for testing purposes

Add bacula-restores task to symlink backup files to bacula restore folder

Comment out restore file permission tasks:
- can't set permissions on symlinks

---
diff --git a/.gitignore b/.gitignore
index e69de29b..c1a0118f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -0,0 +1 @@
+/.vagrant/
diff --git a/Vagrantfile b/Vagrantfile
new file mode 100644
index 00000000..a124020d
--- /dev/null
+++ b/Vagrantfile
@@ -0,0 +1,96 @@
+$bbworker = <<-SCRIPT
+yum install -y \
+    epel-release
+
+yum install -y \
+    device-mapper-persistent-data \
+    git \
+    lvm2 \
+    python-pip \
+    yum-utils
+
+yum-config-manager \
+  --add-repo \
+  https://download.docker.com/linux/centos/docker-ce.repo
+
+yum remove -y \
+    docker \
+    docker-client \
+    docker-client-latest \
+    docker-common \
+    docker-latest \
+    docker-latest-logrotate \
+    docker-logrotate
+
+yum install -y \
+    docker-ce
+
+pip install \
+    docker
+
+systemctl start docker
+docker run --rm hello-world
+SCRIPT
+
+Vagrant.configure("2") do |config|
+
+    config.vm.define "vmdaedalus", primary: true do |main|
+        main.vm.box = "daedalus"
+        main.vm.box_url = "vmdaedalus/vmdaedalus-virtualbox.box"
+        main.vm.hostname = "vmdaedalus.novatech-llc.com"
+        main.vm.provider "virtualbox" do |vb|
+            vb.name = "vmdaedalus"
+            vb.memory = 6000
+            vb.cpus = 4
+        end
+    #    config.vm.network "public_network", ip: "192.168.0.100", netmask:"255.255.0.0", bridge: "enp0s31f6"
+        main.vm.network "private_network", 
+            ip: "192.168.0.100", 
+            netmask:"255.255.0.0"
+
+        main.vm.network "forwarded_port",
+            guest: 2375,
+            host: 12375,
+            protocol: "tcp",
+            id: "docker"
+
+        main.vm.network "forwarded_port", 
+            guest: 22, 
+            host: 10022,
+            id: "ssh"
+
+        main.vm.synced_folder "/mnt/vm/bacula-restores",
+            "/mnt/bacula-storage",
+            id: "bacula-storage",
+            owner: "root",
+            group: "root",
+            mount_options: ["ro"]
+    end
+
+    config.vm.define "bbworker" do |worker|
+        worker.vm.box = "centos/7"
+        worker.vm.hostname = "bbworker.novatech-llc.com"
+        worker.vm.provider "virtualbox" do |vb|
+            vb.name = "bbworker"
+            vb.cpus = 1
+        end
+        
+        worker.vm.network "private_network", 
+            ip: "192.168.0.150", 
+            netmask:"255.255.0.0"
+
+        worker.vm.network "forwarded_port",
+            guest: 2375,
+            host: 22375,
+            protocol: "tcp",
+            id: "docker"
+
+        worker.vm.network "forwarded_port", 
+            guest: 22, 
+            host: 20022,
+            id: "ssh"
+
+        worker.vm.provision "shell", inline: $bbworker
+
+    end
+end
diff --git a/ansible-playbook/group_vars/docker-hosts/docker.yml b/ansible-playbook/group_vars/docker-hosts/docker.yml
index 4623feba..f9153cdc 100644
--- a/ansible-playbook/group_vars/docker-hosts/docker.yml
+++ b/ansible-playbook/group_vars/docker-hosts/docker.yml
@@ -17,7 +17,7 @@ docker_restore_config_base_dir: '/etc/docker_container.conf/{{ docker_image_tag
 docker_backup_dir: '/tmp/docker'
 
 # docker service restart policy
-docker_restart_policy: 'always'
+docker_restart_policy: 'no'
 
 # Network names
 docker_network_frontend: 'frontend'
diff --git a/ansible-playbook/group_vars/docker-hosts/repos.yml b/ansible-playbook/group_vars/docker-hosts/repos.yml
new file mode 100644
index 00000000..cc4d9cbf
--- /dev/null
+++ b/ansible-playbook/group_vars/docker-hosts/repos.yml
@@ -0,0 +1,13 @@
+---
+
+buildbot_image_repo: https://github.com/AndrewCooper/docker-buildbot-master.git
+buildbot_worker_ntel_image_repo: https://github.com/AndrewCooper/docker-buildbot-worker-ntel.git
+buildbot_worker_ptxdist_image_repo: https://github.com/AndrewCooper/docker-buildbot-worker-ptxdist.git
+exim4_image_repo: https://github.com/AndrewCooper/docker-exim4.git
+gitlab_image_repo: 'https://github.com/AndrewCooper/docker-gitlab.git'
+mantisbt_image_repo: https://github.com/AndrewCooper/docker-mantisbt.git
+openldap_image_repo: https://github.com/AndrewCooper/docker-openldap.git
+openssl_image_repo: https://github.com/AndrewCooper/docker-openssl.git
+phpldapadmin_image_repo: https://github.com/AndrewCooper/docker-phpldapadmin.git
+svn_image_repo: https://github.com/AndrewCooper/docker-svn.git
+wiki_image_repo: https://github.com/AndrewCooper/docker-mediawiki.git
diff --git a/ansible-playbook/host_vars/buildbot-worker.novatech-llc.com/privileges.yml b/ansible-playbook/host_vars/buildbot-worker.novatech-llc.com/privileges.yml
new file mode 100644
index 00000000..fa188ba9
--- /dev/null
+++ b/ansible-playbook/host_vars/buildbot-worker.novatech-llc.com/privileges.yml
@@ -0,0 +1,3 @@
+---
+
+ansible_user: vagrant
diff --git a/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
new file mode 100644
index 00000000..ccbedf8a
--- /dev/null
+++ b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
@@ -0,0 +1,18 @@
+---
+
+ansible_user: vagrant
+
+daedalus_subnet: 192.168.0.0/16
+daedalus_net_connection: "enp5s0f0"
+
+svn_ip_addr:          192.168.0.101
+gitlab_ip_addr:       192.168.0.102
+wiki_ip_addr:         192.168.0.103
+mantisbt_ip_addr:     192.168.0.104
+build_ip_addr:        192.168.0.105
+testssh_ip_addr:      192.168.0.106
+exim4_ip_addr:        192.168.0.107
+openldap_ip_addr:     192.168.0.108
+phpldapadmin_ip_addr: 192.168.0.108
+phpmyadmin_ip_addr:   192.168.0.109
+buildbot_ip_addr:     192.168.0.110
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index b839e1e3..de8660ab 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -1,6 +1,18 @@
 ---
 # This playbook deploys the whole application stack in this site.
 
+- name: Dump Variables
+  hosts: all
+  tags:
+    - debugging
+  tasks:
+  - template:
+      src: templates/dump_variables
+      dest: /tmp/ansible_variables
+  - fetch:
+      src: /tmp/ansible_variables
+      dest: "{{inventory_hostname}}_ansible_variables"
+
 - name: Bacula
   hosts: bacula
   remote_user: ansibleremote
@@ -17,7 +29,7 @@
   - role: bacula-console
   - role: bacula-script
     assemble_scripts: false
-  - role: bacula-catalog
+  # - role: bacula-catalog
 
 - name: Prepare Docker host systems
   hosts: docker-hosts
@@ -30,6 +42,10 @@
     role: sshd
     tags:
     - sshd
+  - name: Run bacula-restores symlinks
+    role: bacula-restores
+    tags:
+    - bacula_restores
   - name: Run common Docker tasks
     role: docker-common
     tags:
diff --git a/ansible-playbook/staging b/ansible-playbook/staging
index fa054a69..9d77e053 100644
--- a/ansible-playbook/staging
+++ b/ansible-playbook/staging
@@ -1,3 +1,5 @@
+buildbot-worker.novatech-llc.com ansible_host=127.0.0.1 ansible_port=20022 ansible_user='vagrant' ansible_ssh_private_key_file='/home/coopera/Developer/testdaedalus/.vagrant/machines/bbworker/virtualbox/private_key'
+
 [bacula]
 testdaedalus.novatech-llc.com
 
diff --git a/ansible-playbook/templates/dump_variables b/ansible-playbook/templates/dump_variables
new file mode 100644
index 00000000..3c11ab92
--- /dev/null
+++ b/ansible-playbook/templates/dump_variables
@@ -0,0 +1,7 @@
+HOSTVARS (ANSIBLE GATHERED, group_vars, host_vars) :
+
+{{ hostvars[inventory_hostname] | to_yaml }}
+
+PLAYBOOK VARS:
+
+{{ vars | to_yaml }}
diff --git a/ansible-playbook/vmdaedalus b/ansible-playbook/vmdaedalus
new file mode 100644
index 00000000..dc2fd1e9
--- /dev/null
+++ b/ansible-playbook/vmdaedalus
@@ -0,0 +1,15 @@
+vmdaedalus.novatech-llc.com ansible_host=127.0.0.1 ansible_port=10022 ansible_user='vagrant' ansible_ssh_private_key_file='/home/coopera/Developer/testdaedalus/.vagrant/machines/vmdaedalus/virtualbox/private_key'
+buildbot-worker.novatech-llc.com ansible_host=127.0.0.1 ansible_port=20022 ansible_user='vagrant' ansible_ssh_private_key_file='/home/coopera/Developer/testdaedalus/.vagrant/machines/bbworker/virtualbox/private_key'
+
+[bacula]
+vmdaedalus.novatech-llc.com
+
+[daedalus]
+vmdaedalus.novatech-llc.com
+
+[docker-hosts]
+vmdaedalus.novatech-llc.com
+buildbot-worker.novatech-llc.com
+
+[buildbot-worker]
+buildbot-worker.novatech-llc.com
diff --git a/ansible.cfg b/ansible.cfg
new file mode 100644
index 00000000..7dbeeeb4
--- /dev/null
+++ b/ansible.cfg
@@ -0,0 +1,477 @@
+# config file for ansible -- https://ansible.com/
+# ===============================================
+
+# nearly all parameters can be overridden in ansible-playbook
+# or with command line flags. ansible will read ANSIBLE_CONFIG,
+# ansible.cfg in the current working directory, .ansible.cfg in
+# the home directory or /etc/ansible/ansible.cfg, whichever it
+# finds first
+
+[defaults]
+
+# some basic default values...
+
+#inventory      = /etc/ansible/hosts
+#library        = /usr/share/my_modules/
+#module_utils   = /usr/share/my_module_utils/
+#remote_tmp     = ~/.ansible/tmp
+#local_tmp      = ~/.ansible/tmp
+#plugin_filters_cfg = /etc/ansible/plugin_filters.yml
+#forks          = 5
+#poll_interval  = 15
+#sudo_user      = root
+#ask_sudo_pass = True
+#ask_pass      = True
+#transport      = smart
+#remote_port    = 22
+#module_lang    = C
+#module_set_locale = False
+
+# plays will gather facts by default, which contain information about
+# the remote system.
+#
+# smart - gather by default, but don't regather if already gathered
+# implicit - gather by default, turn off with gather_facts: False
+# explicit - do not gather by default, must say gather_facts: True
+#gathering = implicit
+
+# This only affects the gathering done by a play's gather_facts directive,
+# by default gathering retrieves all facts subsets
+# all - gather all subsets
+# network - gather min and network facts
+# hardware - gather hardware facts (longest facts to retrieve)
+# virtual - gather min and virtual facts
+# facter - import facts from facter
+# ohai - import facts from ohai
+# You can combine them using comma (ex: network,virtual)
+# You can negate them using ! (ex: !hardware,!facter,!ohai)
+# A minimal set of facts is always gathered.
+#gather_subset = all
+
+# some hardware related facts are collected
+# with a maximum timeout of 10 seconds. This
+# option lets you increase or decrease that
+# timeout to something more suitable for the
+# environment.
+# gather_timeout = 10
+
+# additional paths to search for roles in, colon separated
+#roles_path    = /etc/ansible/roles
+roles_path    = /home/coopera/Developer/testdaedalus/roles
+
+# uncomment this to disable SSH key host checking
+#host_key_checking = False
+
+# change the default callback, you can only have one 'stdout' type  enabled at a time.
+#stdout_callback = skippy
+stdout_callback=debug
+
+## Ansible ships with some plugins that require whitelisting,
+## this is done to avoid running all of a type by default.
+## These setting lists those that you want enabled for your system.
+## Custom plugins should not need this unless plugin author specifies it.
+
+# enable callback plugins, they can output to stdout but cannot be 'stdout' type.
+#callback_whitelist = timer, mail
+
+# Determine whether includes in tasks and handlers are "static" by
+# default. As of 2.0, includes are dynamic by default. Setting these
+# values to True will make includes behave more like they did in the
+# 1.x versions.
+#task_includes_static = False
+#handler_includes_static = False
+
+# Controls if a missing handler for a notification event is an error or a warning
+#error_on_missing_handler = True
+
+# change this for alternative sudo implementations
+#sudo_exe = sudo
+
+# What flags to pass to sudo
+# WARNING: leaving out the defaults might create unexpected behaviours
+#sudo_flags = -H -S -n
+
+# SSH timeout
+#timeout = 10
+
+# default user to use for playbooks if user is not specified
+# (/usr/bin/ansible will use current user as default)
+#remote_user = root
+
+# logging is off by default unless this path is defined
+# if so defined, consider logrotate
+#log_path = /var/log/ansible.log
+
+# default module name for /usr/bin/ansible
+#module_name = command
+
+# use this shell for commands executed under sudo
+# you may need to change this to bin/bash in rare instances
+# if sudo is constrained
+#executable = /bin/sh
+
+# if inventory variables overlap, does the higher precedence one win
+# or are hash values merged together?  The default is 'replace' but
+# this can also be set to 'merge'.
+#hash_behaviour = replace
+
+# by default, variables from roles will be visible in the global variable
+# scope. To prevent this, the following option can be enabled, and only
+# tasks and handlers within the role will see the variables there
+#private_role_vars = yes
+
+# list any Jinja2 extensions to enable here:
+#jinja2_extensions = jinja2.ext.do,jinja2.ext.i18n
+
+# if set, always use this private key file for authentication, same as
+# if passing --private-key to ansible or ansible-playbook
+#private_key_file = /path/to/file
+
+# If set, configures the path to the Vault password file as an alternative to
+# specifying --vault-password-file on the command line.
+#vault_password_file = /path/to/vault_password_file
+
+# format of string {{ ansible_managed }} available within Jinja2
+# templates indicates to users editing templates files will be replaced.
+# replacing {file}, {host} and {uid} and strftime codes with proper values.
+#ansible_managed = Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S by {uid} on {host}
+# {file}, {host}, {uid}, and the timestamp can all interfere with idempotence
+# in some situations so the default is a static string:
+#ansible_managed = Ansible managed
+
+# by default, ansible-playbook will display "Skipping [host]" if it determines a task
+# should not be run on a host.  Set this to "False" if you don't want to see these "Skipping"
+# messages. NOTE: the task header will still be shown regardless of whether or not the
+# task is skipped.
+#display_skipped_hosts = True
+
+# by default, if a task in a playbook does not include a name: field then
+# ansible-playbook will construct a header that includes the task's action but
+# not the task's args.  This is a security feature because ansible cannot know
+# if the *module* considers an argument to be no_log at the time that the
+# header is printed.  If your environment doesn't have a problem securing
+# stdout from ansible-playbook (or you have manually specified no_log in your
+# playbook on all of the tasks where you have secret information) then you can
+# safely set this to True to get more informative messages.
+#display_args_to_stdout = False
+
+# by default (as of 1.3), Ansible will raise errors when attempting to dereference
+# Jinja2 variables that are not set in templates or action lines. Uncomment this line
+# to revert the behavior to pre-1.3.
+#error_on_undefined_vars = False
+
+# by default (as of 1.6), Ansible may display warnings based on the configuration of the
+# system running ansible itself. This may include warnings about 3rd party packages or
+# other conditions that should be resolved if possible.
+# to disable these warnings, set the following value to False:
+#system_warnings = True
+
+# by default (as of 1.4), Ansible may display deprecation warnings for language
+# features that should no longer be used and will be removed in future versions.
+# to disable these warnings, set the following value to False:
+#deprecation_warnings = True
+
+# (as of 1.8), Ansible can optionally warn when usage of the shell and
+# command module appear to be simplified by using a default Ansible module
+# instead.  These warnings can be silenced by adjusting the following
+# setting or adding warn=yes or warn=no to the end of the command line
+# parameter string.  This will for example suggest using the git module
+# instead of shelling out to the git command.
+# command_warnings = False
+
+
+# set plugin path directories here, separate with colons
+#action_plugins     = /usr/share/ansible/plugins/action
+#cache_plugins      = /usr/share/ansible/plugins/cache
+#callback_plugins   = /usr/share/ansible/plugins/callback
+#connection_plugins = /usr/share/ansible/plugins/connection
+#lookup_plugins     = /usr/share/ansible/plugins/lookup
+#inventory_plugins  = /usr/share/ansible/plugins/inventory
+#vars_plugins       = /usr/share/ansible/plugins/vars
+#filter_plugins     = /usr/share/ansible/plugins/filter
+#test_plugins       = /usr/share/ansible/plugins/test
+#terminal_plugins   = /usr/share/ansible/plugins/terminal
+#strategy_plugins   = /usr/share/ansible/plugins/strategy
+
+
+# by default, ansible will use the 'linear' strategy but you may want to try
+# another one
+#strategy = free
+
+# by default callbacks are not loaded for /bin/ansible, enable this if you
+# want, for example, a notification or logging callback to also apply to
+# /bin/ansible runs
+#bin_ansible_callbacks = False
+
+
+# don't like cows?  that's unfortunate.
+# set to 1 if you don't want cowsay support or export ANSIBLE_NOCOWS=1
+#nocows = 1
+
+# set which cowsay stencil you'd like to use by default. When set to 'random',
+# a random stencil will be selected for each task. The selection will be filtered
+# against the `cow_whitelist` option below.
+#cow_selection = default
+#cow_selection = random
+
+# when using the 'random' option for cowsay, stencils will be restricted to this list.
+# it should be formatted as a comma-separated list with no spaces between names.
+# NOTE: line continuations here are for formatting purposes only, as the INI parser
+#       in python does not support them.
+#cow_whitelist=bud-frogs,bunny,cheese,daemon,default,dragon,elephant-in-snake,elephant,eyes,\
+#              hellokitty,kitty,luke-koala,meow,milk,moofasa,moose,ren,sheep,small,stegosaurus,\
+#              stimpy,supermilker,three-eyes,turkey,turtle,tux,udder,vader-koala,vader,www
+
+# don't like colors either?
+# set to 1 if you don't want colors, or export ANSIBLE_NOCOLOR=1
+#nocolor = 1
+
+# if set to a persistent type (not 'memory', for example 'redis') fact values
+# from previous runs in Ansible will be stored.  This may be useful when
+# wanting to use, for example, IP information from one group of servers
+# without having to talk to them in the same playbook run to get their
+# current IP information.
+#fact_caching = memory
+
+
+# retry files
+# When a playbook fails by default a .retry file will be created in ~/
+# You can disable this feature by setting retry_files_enabled to False
+# and you can change the location of the files by setting retry_files_save_path
+
+#retry_files_enabled = False
+#retry_files_save_path = ~/.ansible-retry
+
+# squash actions
+# Ansible can optimise actions that call modules with list parameters
+# when looping. Instead of calling the module once per with_ item, the
+# module is called once with all items at once. Currently this only works
+# under limited circumstances, and only with parameters named 'name'.
+#squash_actions = apk,apt,dnf,homebrew,pacman,pkgng,yum,zypper
+
+# prevents logging of task data, off by default
+#no_log = False
+
+# prevents logging of tasks, but only on the targets, data is still logged on the master/controller
+#no_target_syslog = False
+
+# controls whether Ansible will raise an error or warning if a task has no
+# choice but to create world readable temporary files to execute a module on
+# the remote machine.  This option is False by default for security.  Users may
+# turn this on to have behaviour more like Ansible prior to 2.1.x.  See
+# https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user
+# for more secure ways to fix this than enabling this option.
+#allow_world_readable_tmpfiles = False
+
+# controls the compression level of variables sent to
+# worker processes. At the default of 0, no compression
+# is used. This value must be an integer from 0 to 9.
+#var_compression_level = 9
+
+# controls what compression method is used for new-style ansible modules when
+# they are sent to the remote system.  The compression types depend on having
+# support compiled into both the controller's python and the client's python.
+# The names should match with the python Zipfile compression types:
+# * ZIP_STORED (no compression. available everywhere)
+# * ZIP_DEFLATED (uses zlib, the default)
+# These values may be set per host via the ansible_module_compression inventory
+# variable
+#module_compression = 'ZIP_DEFLATED'
+
+# This controls the cutoff point (in bytes) on --diff for files
+# set to 0 for unlimited (RAM may suffer!).
+#max_diff_size = 1048576
+
+# This controls how ansible handles multiple --tags and --skip-tags arguments
+# on the CLI.  If this is True then multiple arguments are merged together.  If
+# it is False, then the last specified argument is used and the others are ignored.
+# This option will be removed in 2.8.
+#merge_multiple_cli_flags = True
+
+# Controls showing custom stats at the end, off by default
+#show_custom_stats = True
+
+# Controls which files to ignore when using a directory as inventory with
+# possibly multiple sources (both static and dynamic)
+#inventory_ignore_extensions = ~, .orig, .bak, .ini, .cfg, .retry, .pyc, .pyo
+
+# This family of modules use an alternative execution path optimized for network appliances
+# only update this setting if you know how this works, otherwise it can break module execution
+#network_group_modules=eos, nxos, ios, iosxr, junos, vyos
+
+# When enabled, this option allows lookups (via variables like {{lookup('foo')}} or when used as
+# a loop with `with_foo`) to return data that is not marked "unsafe". This means the data may contain
+# jinja2 templating language which will be run through the templating engine.
+# ENABLING THIS COULD BE A SECURITY RISK
+#allow_unsafe_lookups = False
+
+# set default errors for all plays
+#any_errors_fatal = False
+
+[inventory]
+# enable inventory plugins, default: 'host_list', 'script', 'yaml', 'ini'
+#enable_plugins = host_list, virtualbox, yaml, constructed
+
+# ignore these extensions when parsing a directory as inventory source
+#ignore_extensions = .pyc, .pyo, .swp, .bak, ~, .rpm, .md, .txt, ~, .orig, .ini, .cfg, .retry
+
+# ignore files matching these patterns when parsing a directory as inventory source
+#ignore_patterns=
+
+# If 'true' unparsed inventory sources become fatal errors, they are warnings otherwise.
+#unparsed_is_failed=False
+
+[privilege_escalation]
+#become=True
+#become_method=sudo
+#become_user=root
+#become_ask_pass=False
+
+[paramiko_connection]
+
+# uncomment this line to cause the paramiko connection plugin to not record new host
+# keys encountered.  Increases performance on new host additions.  Setting works independently of the
+# host key checking setting above.
+#record_host_keys=False
+
+# by default, Ansible requests a pseudo-terminal for commands executed under sudo. Uncomment this
+# line to disable this behaviour.
+#pty=False
+
+# paramiko will default to looking for SSH keys initially when trying to
+# authenticate to remote devices.  This is a problem for some network devices
+# that close the connection after a key failure.  Uncomment this line to
+# disable the Paramiko look for keys function
+#look_for_keys = False
+
+# When using persistent connections with Paramiko, the connection runs in a
+# background process.  If the host doesn't already have a valid SSH key, by
+# default Ansible will prompt to add the host key.  This will cause connections
+# running in background processes to fail.  Uncomment this line to have
+# Paramiko automatically add host keys.
+#host_key_auto_add = True
+
+[ssh_connection]
+
+# ssh arguments to use
+# Leaving off ControlPersist will result in poor performance, so use
+# paramiko on older platforms rather than removing it, -C controls compression use
+#ssh_args = -C -o ControlMaster=auto -o ControlPersist=60s
+
+# The base directory for the ControlPath sockets.
+# This is the "%(directory)s" in the control_path option
+#
+# Example:
+# control_path_dir = /tmp/.ansible/cp
+#control_path_dir = ~/.ansible/cp
+
+# The path to use for the ControlPath sockets. This defaults to a hashed string of the hostname,
+# port and username (empty string in the config). The hash mitigates a common problem users
+# found with long hostames and the conventional %(directory)s/ansible-ssh-%%h-%%p-%%r format.
+# In those cases, a "too long for Unix domain socket" ssh error would occur.
+#
+# Example:
+# control_path = %(directory)s/%%h-%%r
+#control_path =
+
+# Enabling pipelining reduces the number of SSH operations required to
+# execute a module on the remote server. This can result in a significant
+# performance improvement when enabled, however when using "sudo:" you must
+# first disable 'requiretty' in /etc/sudoers
+#
+# By default, this option is disabled to preserve compatibility with
+# sudoers configurations that have requiretty (the default on many distros).
+#
+#pipelining = False
+pipelining = True
+
+# Control the mechanism for transferring files (old)
+#   * smart = try sftp and then try scp [default]
+#   * True = use scp only
+#   * False = use sftp only
+#scp_if_ssh = smart
+
+# Control the mechanism for transferring files (new)
+# If set, this will override the scp_if_ssh option
+#   * sftp  = use sftp to transfer files
+#   * scp   = use scp to transfer files
+#   * piped = use 'dd' over SSH to transfer files
+#   * smart = try sftp, scp, and piped, in that order [default]
+#transfer_method = smart
+
+# if False, sftp will not use batch mode to transfer files. This may cause some
+# types of file transfer failures impossible to catch however, and should
+# only be disabled if your sftp version has problems with batch mode
+#sftp_batch_mode = False
+
+# The -tt argument is passed to ssh when pipelining is not enabled because sudo 
+# requires a tty by default. 
+#use_tty = True
+
+[persistent_connection]
+
+# Configures the persistent connection timeout value in seconds.  This value is
+# how long the persistent connection will remain idle before it is destroyed.
+# If the connection doesn't receive a request before the timeout value
+# expires, the connection is shutdown. The default value is 30 seconds.
+#connect_timeout = 30
+
+# Configures the persistent connection retry timeout.  This value configures the
+# the retry timeout that ansible-connection will wait to connect
+# to the local domain socket. This value must be larger than the
+# ssh timeout (timeout) and less than persistent connection idle timeout (connect_timeout).
+# The default value is 15 seconds.
+#connect_retry_timeout = 15
+
+# The command timeout value defines the amount of time to wait for a command
+# or RPC call before timing out. The value for the command timeout must
+# be less than the value of the persistent connection idle timeout (connect_timeout)
+# The default value is 10 second.
+#command_timeout = 10
+
+[accelerate]
+#accelerate_port = 5099
+#accelerate_timeout = 30
+#accelerate_connect_timeout = 5.0
+
+# The daemon timeout is measured in minutes. This time is measured
+# from the last activity to the accelerate daemon.
+#accelerate_daemon_timeout = 30
+
+# If set to yes, accelerate_multi_key will allow multiple
+# private keys to be uploaded to it, though each user must
+# have access to the system via SSH to add a new key. The default
+# is "no".
+#accelerate_multi_key = yes
+
+[selinux]
+# file systems that require special treatment when dealing with security context
+# the default behaviour that copies the existing context or uses the user default
+# needs to be changed to use the file system dependent context.
+#special_context_filesystems=nfs,vboxsf,fuse,ramfs,9p
+
+# Set this to yes to allow libvirt_lxc connections to work without SELinux.
+#libvirt_lxc_noseclabel = yes
+
+[colors]
+#highlight = white
+#verbose = blue
+#warn = bright purple
+#error = red
+#debug = dark gray
+#deprecate = purple
+#skip = cyan
+#unreachable = red
+#ok = green
+#changed = yellow
+#diff_add = green
+#diff_remove = red
+#diff_lines = cyan
+
+
+[diff]
+# Always print diff when running ( same as always running with -D/--diff )
+# always = no
+
+# Set how many context lines to show in diff
+# context = 3
diff --git a/provision.sh b/provision.sh
new file mode 100755
index 00000000..3762e89d
--- /dev/null
+++ b/provision.sh
@@ -0,0 +1,14 @@
+#! /bin/bash
+
+#PYTHONUNBUFFERED=1 \
+     
+ANSIBLE_ENABLE_TASK_DEBUGGER=True \
+ANSIBLE_FORCE_COLOR=true \
+ANSIBLE_HOST_KEY_CHECKING=false \
+ANSIBLE_SSH_ARGS='-o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ControlMaster=auto -o ControlPersist=60s' \
+exec ansible-playbook \
+        --connection=ssh \
+        --timeout=30 \
+        --inventory-file=ansible-playbook/staging \
+        -vv \
+        ansible-playbook/site.yml "$@"
diff --git a/roles/bacula-restores/tasks/main.yml b/roles/bacula-restores/tasks/main.yml
new file mode 100644
index 00000000..560f5ba8
--- /dev/null
+++ b/roles/bacula-restores/tasks/main.yml
@@ -0,0 +1,26 @@
+---
+# tasks file for bacula-restores
+
+- name: List bacula-restores
+  find:
+    paths: /mnt/bacula-restores
+    file_type: file
+    recurse: yes
+  register: bacula_restores_list
+
+- name: Create bacula-restores symlinks
+  file:
+    path: '{{ item.path | replace("/mnt/bacula-restores","/tmp/bacula-restores") | dirname }}'
+    state: directory
+    force: yes
+    mode: 0777
+  with_items: '{{ bacula_restores_list.files }}'
+
+- name: Create bacula-restores directories
+  file:
+    src: '{{ item.path }}'
+    dest: '{{ item.path | replace("/mnt/bacula-restores","/tmp/bacula-restores") }}'
+    state: link
+    force: yes
+    mode: '{{ item.mode }}'
+  with_items: '{{ bacula_restores_list.files }}'
diff --git a/roles/docker-buildsystem/tasks/restore.yml b/roles/docker-buildsystem/tasks/restore.yml
index be79a681..26766490 100644
--- a/roles/docker-buildsystem/tasks/restore.yml
+++ b/roles/docker-buildsystem/tasks/restore.yml
@@ -23,14 +23,14 @@
     path: '{{ restore_buildsystem_docker_dir }}'
   register: buildsystem_restore_files
 
-- name: File permissions
-  file:
-    path: '{{ restore_buildsystem_docker_dir }}/{{ item }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rwx,g=rwx,o='
-  with_items: '{{ buildsystem_restore_files.file_list | default([]) }}'
+# - name: File permissions
+#   file:
+#     path: '{{ restore_buildsystem_docker_dir }}/{{ item }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rwx,g=rwx,o='
+#   with_items: '{{ buildsystem_restore_files.file_list | default([]) }}'
 
 # *****************************************************************************
 # For old backups that included the mysql database.
diff --git a/roles/docker-exim4/tasks/restore.yml b/roles/docker-exim4/tasks/restore.yml
index cf94cf73..69102432 100644
--- a/roles/docker-exim4/tasks/restore.yml
+++ b/roles/docker-exim4/tasks/restore.yml
@@ -20,13 +20,13 @@
     path_to_restore: '{{ exim4_docker_backup_dir }}'
   when: st_exim4_backup_file.stat.exists == False
 
-- name: Set permissions for data-volume data
-  file:
-    path: '{{ exim4_docker_restore_dir }}{{ exim4_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
+# - name: Set permissions for data-volume data
+#   file:
+#     path: '{{ exim4_docker_restore_dir }}{{ exim4_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
 
 - name: exists - exim4 archive
   stat:
diff --git a/roles/docker-gitlab/tasks/restore.yml b/roles/docker-gitlab/tasks/restore.yml
index f553ff0c..fc4cc59c 100644
--- a/roles/docker-gitlab/tasks/restore.yml
+++ b/roles/docker-gitlab/tasks/restore.yml
@@ -23,14 +23,14 @@
     path: '{{ gitlab_docker_restore_dir }}'
   register: gitlab_backup
 
-- name: File permissions
-  file:
-    path: '{{ gitlab_docker_restore_dir }}/{{ item }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rwx,g=rwx,o='
-  with_items: '{{ gitlab_backup.file_list | default([]) }}'
+# - name: File permissions
+#   file:
+#     path: '{{ gitlab_docker_restore_dir }}/{{ item }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rwx,g=rwx,o='
+#   with_items: '{{ gitlab_backup.file_list | default([]) }}'
 
 # *****************************************************************************
 # restore the gitlap backup
diff --git a/roles/docker-mantisbt/tasks/restore.yml b/roles/docker-mantisbt/tasks/restore.yml
index 6289ba78..78f66b97 100644
--- a/roles/docker-mantisbt/tasks/restore.yml
+++ b/roles/docker-mantisbt/tasks/restore.yml
@@ -34,29 +34,29 @@
     path_to_restore: '{{ mantisbt_docker_backup_dir }}'
   when: st_mantisbt_static_archive.stat.exists == False or st_mantisbt_SQL_file.stat.exists == False or st_mantisbt_files_table_SQL_file.stat.exists == False
 
-- name: permissions - Static
-  file:
-    path: '{{ mantisbt_docker_restore_dir }}{{ mantisbt_static_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
-
-- name: permissions - SQL (files table)
-  file:
-    path: '{{ mantisbt_docker_restore_dir }}{{ mantisbt_database_files_table_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
-
-- name: permissions - SQL (remaining tables)
-  file:
-    path: '{{ mantisbt_docker_restore_dir }}{{ mantisbt_database_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
+# - name: permissions - Static
+#   file:
+#     path: '{{ mantisbt_docker_restore_dir }}{{ mantisbt_static_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
+
+# - name: permissions - SQL (files table)
+#   file:
+#     path: '{{ mantisbt_docker_restore_dir }}{{ mantisbt_database_files_table_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
+
+# - name: permissions - SQL (remaining tables)
+#   file:
+#     path: '{{ mantisbt_docker_restore_dir }}{{ mantisbt_database_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
 
 - name: exists - Static
   stat:
diff --git a/roles/docker-mediawiki/tasks/restore.yml b/roles/docker-mediawiki/tasks/restore.yml
index 6ff541ed..f2b3409a 100644
--- a/roles/docker-mediawiki/tasks/restore.yml
+++ b/roles/docker-mediawiki/tasks/restore.yml
@@ -27,21 +27,21 @@
     path_to_restore: '{{ wiki_docker_backup_dir }}'
   when: st_wiki_static_archive.stat.exists == False or st_wiki_SQL_file.stat.exists == False
 
-- name: permissions - Static
-  file:
-    path: '{{ wiki_docker_restore_dir }}{{ wiki_static_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
+# - name: permissions - Static
+#   file:
+#     path: '{{ wiki_docker_restore_dir }}{{ wiki_static_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
 
-- name: permissions - SQL
-  file:
-    path: '{{ wiki_docker_restore_dir }}{{ wiki_database_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
+# - name: permissions - SQL
+#   file:
+#     path: '{{ wiki_docker_restore_dir }}{{ wiki_database_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
 
 - name: exists - Static
   stat:
diff --git a/roles/docker-openldap/tasks/restore.yml b/roles/docker-openldap/tasks/restore.yml
index 8fb5ac24..feaf93e7 100644
--- a/roles/docker-openldap/tasks/restore.yml
+++ b/roles/docker-openldap/tasks/restore.yml
@@ -18,14 +18,14 @@
     path_to_restore: '{{ openldap_docker_backup_dir }}'
   when: openldap_backup.file_list == []
 
-- name: Set permissions for data-volume data
-  file:
-    path: '{{ openldap_docker_restore_dir }}/{{ item.ldif_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
-  with_items: '{{ openldap_backup_files }}'
+# - name: Set permissions for data-volume data
+#   file:
+#     path: '{{ openldap_docker_restore_dir }}/{{ item.ldif_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
+#   with_items: '{{ openldap_backup_files }}'
 
 # *****************************************************************************
 # restore the openldap backup
diff --git a/roles/docker-openssl/tasks/restore.yml b/roles/docker-openssl/tasks/restore.yml
index ba36c6f3..1cfc3d4f 100644
--- a/roles/docker-openssl/tasks/restore.yml
+++ b/roles/docker-openssl/tasks/restore.yml
@@ -20,13 +20,13 @@
     path_to_restore: '{{ openssl_docker_backup_dir }}'
   when: st_openssl_backup_file.stat.exists == False
 
-- name: Set permissions for data-volume data
-  file:
-    path: '{{ openssl_docker_restore_dir }}{{ openssl_backup_file }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rw,g=rw,o='
+# - name: Set permissions for data-volume data
+#   file:
+#     path: '{{ openssl_docker_restore_dir }}{{ openssl_backup_file }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rw,g=rw,o='
 
 - name: exists - openssl archive
   stat:
diff --git a/roles/docker-svn/tasks/restore.yml b/roles/docker-svn/tasks/restore.yml
index 0a3fa25a..22e72047 100644
--- a/roles/docker-svn/tasks/restore.yml
+++ b/roles/docker-svn/tasks/restore.yml
@@ -23,14 +23,14 @@
     path: '{{ svn_docker_restore_dir }}'
   register: svn_repo_backup
 
-- name: File permissions
-  file:
-    path: '{{ svn_docker_restore_dir }}/{{ item }}'
-    state: touch
-    owner: root
-    group: root
-    mode: 'u=rwx,g=rwx,o='
-  with_items: '{{ svn_repo_backup.file_list | default([]) }}'
+# - name: File permissions
+#   file:
+#     path: '{{ svn_docker_restore_dir }}/{{ item }}'
+#     state: touch
+#     owner: root
+#     group: root
+#     mode: 'u=rwx,g=rwx,o='
+#   with_items: '{{ svn_repo_backup.file_list | default([]) }}'
 
 # *****************************************************************************
 # restore the svn backup
diff --git a/vmdaedalus/.gitignore b/vmdaedalus/.gitignore
new file mode 100644
index 00000000..c148b6f5
--- /dev/null
+++ b/vmdaedalus/.gitignore
@@ -0,0 +1,2 @@
+bacula-restores/
+vmdaedalus-virtualbox.box
diff --git a/vmdaedalus/vmdaedalus.cfg b/vmdaedalus/vmdaedalus.cfg
new file mode 100644
index 00000000..5d5fdee4
--- /dev/null
+++ b/vmdaedalus/vmdaedalus.cfg
@@ -0,0 +1,71 @@
+install
+text
+cdrom
+skipx
+lang en_US.UTF-8
+keyboard us
+timezone UTC
+rootpw vagrant
+user --name=vagrant --password=vagrant
+auth --enableshadow --passalgo=sha512 --kickstart
+firewall --disabled
+selinux --permissive
+bootloader --location=mbr
+zerombr
+clearpart --all --initlabel
+autopart
+firstboot --disable
+reboot
+repo --name=epel   --baseurl=http://download.fedoraproject.org/pub/epel/7/x86_64/
+repo --name=extras --baseurl=http://mirror.centos.org/centos/7/extras/x86_64/
+repo --name=docker --baseurl=http://download.docker.com/linux/centos/7/x86_64/stable/
+
+%packages --instLangs=en_US.utf8 --nobase --ignoremissing --excludedocs
+@base
+@core
+@hardware-monitoring
+@input-methods
+@performance
+@mariadb
+@mariadb-client
+epel-release
+bacula-client
+bacula-common
+bacula-console
+bacula-director
+bacula-libs
+bacula-libs-sql
+bacula-storage
+bacula-console-bat
+docker-ce
+git
+subversion
+pexpect
+yum-utils
+%end
+ 
+ 
+%post --log=/root/ks.log
+ 
+echo "vagrant ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers.d/vagrant
+echo "Defaults:vagrant !requiretty" >> /etc/sudoers.d/vagrant
+chmod 0440 /etc/sudoers.d/vagrant
+mkdir -pm 700 /home/vagrant/.ssh
+#curl -o /home/vagrant/.ssh/authorized_keys https://raw.githubusercontent.com/mitchellh/vagrant/master/keys/vagrant.pub
+cat <<EOK >/home/vagrant/.ssh/authorized_keys
+ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8Y\
+Vr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdO\
+KLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7Pt\
+ixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmC\
+P3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcW\
+yLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key
+EOK
+chmod 0600 /home/vagrant/.ssh/authorized_keys
+chown -R vagrant.vagrant /home/vagrant/.ssh
+yum -y update
+yum -y remove linux-firmware
+yum-config-manager \
+    --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo
+
+%end
diff --git a/vmdaedalus/vmdaedalus.json b/vmdaedalus/vmdaedalus.json
new file mode 100644
index 00000000..2703db83
--- /dev/null
+++ b/vmdaedalus/vmdaedalus.json
@@ -0,0 +1,43 @@
+{
+    "variables": {
+    },
+    "builders": [{
+        "type": "virtualbox-iso",
+        "guest_os_type": "RedHat_64",
+        "iso_url": "/home/coopera/Downloads/CentOS-7-x86_64-DVD-1708/CentOS-7-x86_64-DVD-1708.iso",
+        "iso_checksum": "ec7500d4b006702af6af023b1f8f1b890b6c7ee54400bb98cef968b883cd6546",
+        "iso_checksum_type": "sha256",
+        "ssh_username": "vagrant",
+        "ssh_password": "vagrant",
+        "ssh_wait_timeout": "600s",
+        "vm_name": "vmdaedalus",
+        "http_directory": "./",
+        "boot_wait": "10s",
+        "boot_command": [
+            "<esc><wait>",
+            "linux ks=http://{{.HTTPIP}}:{{.HTTPPort}}/vmdaedalus.cfg",
+            "<enter><wait>"
+        ],
+        "shutdown_command": "echo 'vagrant' |sudo -S shutdown -P now",
+        "vboxmanage": [
+            ["modifyvm", "{{.Name}}", "--memory", "512"],
+            ["modifyvm", "{{.Name}}", "--usb", "off"],
+            ["modifyvm", "{{.Name}}", "--audio", "none"]
+        ],
+        "disk_size": "120000"
+    }],
+    "provisioners": [{
+        "type": "shell",
+        "inline": [
+            "sudo yum -y install gcc make bzip2 kernel-headers kernel-devel dkms",
+            "sudo yum clean all",
+            "sudo mount -o loop /home/vagrant/VBoxGuestAdditions.iso /mnt",
+            "sudo sh /mnt/VBoxLinuxAdditions.run",
+            "sudo umount /mnt"
+        ]
+    }],
+    "post-processors": [{
+        "type": "vagrant",
+        "output": "./vmdaedalus-{{.Provider}}.box"
+    }]
+}
